src/Resources/config/core.php

```diff
@@ -151,6 +152,23 @@
 
         ->alias(SearchHandler::class, 'sonata.admin.search.handler')
 
+        ->set('sonata.admin.controller.crud', CRUDController::class)
+            ->public()
+            ->tag('container.service_subscriber')
+            ->args([
+                new ReferenceConfigurator('sonata.admin.pool'),
+                new ReferenceConfigurator('sonata.admin.audit.manager'),
+                new ReferenceConfigurator('sonata.admin.object.manipulator.acl.admin'),
+                new ReferenceConfigurator('sonata.admin.breadcrumbs_builder'),
+                new ReferenceConfigurator('request_stack'),
+                new ReferenceConfigurator('translator'),
+                (new ReferenceConfigurator('sonata.exporter.exporter'))->nullOnInvalid(),
+                (new ReferenceConfigurator('sonata.admin.admin_exporter'))->nullOnInvalid(),
+                (new ReferenceConfigurator('sonata.admin.security.acl_user_manager'))->nullOnInvalid(),
+                (new ReferenceConfigurator('logger'))->nullOnInvalid(),
+            ])
+            ->call('setContainer', [new ReferenceConfigurator('service_container')])
```

Comment:
Do we need to keep `setContainer` ?

Reply:
Read my comments above this, the crud controller uses methods like $this->render (from AbstractController), and those methods needs having a Container.